### DISCUSSION TITLE
Feature Request: Support for OwnTracks MQTT mode

### DISCUSSION THREAD

**Discussion by tobru** [Freika/dawarich#267] [Ideas]
It would be wonderful to see MQTT mode of OwnTracks supported in dawarich.

I'm a heavy user of it, and it would make dawarich a pretty awesome companion for the OwnTracks community.

**Comment by Freika** [comment]
Nice idea, might work on it 👍

**Reply by tobru** [reply]
Thanks for your reply and dedication to the project.

**Reply by elyobelyob** [reply]
This would be great, as I only make my MQTT public.

**Reply by mamo2021** [reply]
I would use it too.

**Comment by elyobelyob** [comment]
OK, I decided to just use MQTT in to my network, and then use an existing Node-Red to pass the data onto the Dawarich Owntracks API. Works great.

**Reply by ensjens** [reply]
Would you mind sharing your node-red flow? I have the same use-case.

**Reply by pbek** [reply]
I wonder what happens to the location data if the MQTT server or Node-RED or something else goes away. 🤔

**Reply by elyobelyob** [reply]
It's very simple, an mqtt in node listening for the correct Topic, connected to an http reuest node. That's set to POST and the URL set to http://192.168.0.21:3000/api/v1/owntracks/points?api_key=xxxxx

```
[
    {
        "id": "a9abacc238e05595",
        "type": "tab",
        "label": "Dawarich",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "55f1d08588f9d929",
        "type": "mqtt in",
        "z": "a9abacc238e05595",
        "name": "",
        "topic": "owntracks/#",
        "qos": "2",
        "datatype": "auto-detect",
        "broker": "a7b331d61fc868a5",
        "nl": false,
        "rap": true,
        "rh": 0,
        "inputs": 0,
        "x": 190,
        "y": 160,
        "wires": [
            [
                "9a206529198327ea"
            ]
        ]
    },
    {
        "id": "9a206529198327ea",
        "type": "http request",
        "z": "a9abacc238e05595",
        "name": "",
        "method": "POST",
        "ret": "obj",
        "paytoqs": "body",
        "url": "http://192.168.0.21:3000/api/v1/owntracks/points?api_key=xxxxx",
        "tls": "",
        "persist": false,
        "proxy": "",
        "insecureHTTPParser": false,
        "authType": "",
        "senderr": false,
        "headers": [],
        "x": 450,
        "y": 160,
        "wires": [
            []
        ]
    },
    {
        "id": "ef7a25cf037d6ce9",
        "type": "comment",
        "z": "a9abacc238e05595",
        "name": "Data Format",
        "info": "{\n  \"_type\": \"location\",\n  \"acc\": 5,\n  \"alt\": 12,\n  \"batt\": 100,\n  \"bs\": 3,\n  \"BSSID\": \"90:9a:4a:80:2f:c6\",\n  \"cog\": 315,\n  \"conn\": \"w\",\n  \"lat\": 51.39196,\n  \"lon\": -0.312363,\n  \"m\": 2,\n  \"p\": 102.696,\n  \"SSID\": \"TP-LINK_9743_5G\",\n  \"t\": \"t\",\n  \"tid\": \"nb\",\n  \"tst\": 1739114674,\n  \"vac\": 3,\n  \"vel\": 0\n}",
        "x": 150,
        "y": 80,
        "wires": []
    },
    {
        "id": "a7b331d61fc868a5",
        "type": "mqtt-broker",
        "name": "192.168.0.12",
        "broker": "192.168.0.12",
        "port": "1883",
        "clientid": "",
        "autoConnect": true,
        "usetls": false,
        "protocolVersion": "4",
        "keepalive": "60",
        "cleansession": true,
        "autoUnsubscribe": true,
        "birthTopic": "",
        "birthQos": "0",
        "birthRetain": "false",
        "birthPayload": "",
        "birthMsg": {},
        "closeTopic": "",
        "closeQos": "0",
        "closeRetain": "false",
        "closePayload": "",
        "closeMsg": {},
        "willTopic": "",
        "willQos": "0",
        "willRetain": "false",
        "willPayload": "",
        "willMsg": {},
        "userProps": "",
        "sessionExpiry": ""
    }
]
```

**Reply by elyobelyob** [reply]
> I wonder what happens to the location data if the MQTT server or Node-RED or something else goes away. 🤔

OwnTracks records the data locally until it comes back. Here's what they say about the iOS app ... "We store an unlimited (well, limited by iOS resources only) number of points in an SQLite database on the device, so it survives device reboots, etc."

**Reply by pbek** [reply]
> > I wonder what happens to the location data if the MQTT server or Node-RED or something else goes away. 🤔
> 
> OwnTracks records the data locally until it comes back. Here's what they say about the iOS app ... "We store an unlimited (well, limited by iOS resources only) number of points in an SQLite database on the device, so it survives device reboots, etc."

But that doesn't help if nothing is consuming the messages on mqtt. In mqtt you can only retain the last message.

**Reply by elyobelyob** [reply]
A bit beyond anyone else's control but yours, tbf.

**Reply by pbek** [reply]
Of course, but that is not the point. I'm a heavy user of MQTT, but I wouldn't use it in this context. Because I want to keep track of every data point, not just the last one. So I'd rather collect data locally and do bulk-upload of the data points as soon that network is available than to rely on the uptime of all services and the ability to process the data in real-time... But that's just my personal opinion.

**Reply by elyobelyob** [reply]
But, you can have multiple things listening to the topic locally.

**Reply by pbek** [reply]
Does that help DaWarIch collecting the data?
And what if your phone does not have network connection for a while? 🤔

**Reply by elyobelyob** [reply]
> Does that help DaWarIch collecting the data? 
What happens after it hits MQTT is entirely your decision. I have HA also sending points to Dawarich. I tried having the Dawarich app too, but that didn't suit my needs. Perhaps you want to inject direct into the redis db?

> And what if your phone does not have network connection for a while? 🤔
Already answered that ... it stores it until the MQTT connection is available.

**Comment by alarazr** [comment]
I would 100% use this because my network connectivity between Owntracks and Dawarich is always unreliable, so MQTT would ensure data is always retained

**Comment by pytz** [comment]
Would also love to see that feature!

Just like @elyobelyob (https://github.com/Freika/dawarich/discussions/267#discussioncomment-12174163), @alarazr (https://github.com/Freika/dawarich/discussions/996) and a user over in the OwnTracks repo (https://github.com/owntracks/frontend/issues/160#issuecomment-3093798002) I build myself a small python script to collect MQTT shared locations and POST them to Dawarich.
I prefer MQTT (via OwnTracks app with [client certificate authentication](https://github.com/owntracks/booklet/blob/master/docs/features/tlscert.md)) as it allows multiple applications to react on location updates. In my case otrecorder (used it many years and would like to continue to use it as long as Dawarich is still "under active development" as stated in the disclaimer), Home Assistant (location based automations), Dawarich (amazing frontend features I miss in OwnTracks).

Using OwnTracks recorder as backend for Dawarich would also solve this for me (an idea posted over in https://github.com/owntracks/frontend/issues/160 and partially discussed in https://github.com/Freika/dawarich/discussions/571 or https://github.com/Freika/dawarich/discussions/1346). But this is a different topic and probably much more difficult to implement (if possible and desired at all).
